### ISSUE TITLE
SC2046 does not work with the kill command

### ISSUE THREAD

**Issue by tanducmai** [koalaman/shellcheck#2743]
#### For bugs
- Rule Id (SC2046): 
- My shellcheck version (0.9.0): 
- The rule's wiki page does not already cover this (https://shellcheck.net/wiki/SC2046)
- I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
kill $(pidof nginx)

```

#### Here's what shellcheck currently says:
Quote this to prevent word splitting. [SC2046]

#### Issue
When I quote the highlighted part, as in

```sh

#!/bin/bash
kill "$(pidof nginx)"

```

The error with the command:

```text
kill: failed to parse argument: '2098 2097 2096 2095 2094 2093 2092 2091 2090 2089 2088 2087 2086'
```

#### Expected Result
Nothing should appear on the screen as the process nginx is just silently killed.

**Comment by brother** [comment]
Maybe `kill` might warrant an exception. Or maybe not. 

In the mean time just add a [directive](https://github.com/koalaman/shellcheck/wiki/Directive) to ignore it. You have read and understood the warning and have deemed the code path as a totally valid case and want the word splitting.

**Comment by kurahaupo** [comment]
Using `some_command $(pidof ...)` warrants its own warning, as it introduces a race condition, and always runs `some_command` even if no processes are identified.

Using `pidof` should be a last resort for managing processes.

1. Use your system process manager (systemd, upstart, init, rc, etc)
2. Actively manage your processes: in your "start" scripts, store the process IDs of started jobs; then in your "stop" scripts, retrieve them. (If one script invocation does both, it can use variables; otherwise use files.)
3. To kill unmanaged processes, use `pkill ...` (or its equivalent on your system)
4.  Avoid `kill $(pidof ...)` if at all possible.
